packages/plugin-file-upload/web/src/file-upload-component.jsx

```diff
@@ -63,15 +66,25 @@ class FileUploadComponent extends PureComponent {
       if (name && name.includes('.')) {
         type = name.split('.').pop();
       }
-      this.updateComponentData({ name, type, size: file.size, tempData: true });
+      const size = file.size;
+      this.updateComponentData({ name, type, size, tempData: true });
+      const uploadingFileBI = onUploadStart(
+        FILE_UPLOAD_TYPE,
+        this.helpers?.onMediaUploadStart,
+        size,
+        getMediaType(type)
+      );
       this.setState({ isLoading: true });
-      onFileSelected(file, ({ data, error }) => this.handleFilesAdded({ data, error }));
+      onFileSelected(file, ({ data, error }) =>
+        this.handleFilesAdded({ data, error, uploadingFileBI })
+      );
     } else {
       this.resetLoadingState({ msg: 'missing upload function' });
     }
   };
 
-  handleFilesAdded = ({ data, error }) => {
+  handleFilesAdded = ({ data, error, uploadingFileBI }) => {
+    uploadingFileBI && onUploadEnd(this.props.helpers?.onMediaUpload, uploadingFileBI, error);
```

Comment:
Correct me if I'm wrong - I think `uploadingFileBI` is always `truthy` in this stage (?)
```suggestion
    onUploadEnd(this.props.helpers?.onMediaUpload, uploadingFileBI, error);
```

Reply:
It's actually related to your previous question - currently it's not always truthy since handleFilesAdded can be called from bastToolbarButton directly (difference between native uploads and media manager uploads)
I'm thinking of a way around it